dev/io.openliberty.microprofile.metrics.internal.3.0/src/io/openliberty/microprofile/metrics30/internal/helper/BucketManager.java

```diff
@@ -0,0 +1,298 @@
+
+                    defaultBucketsMicrometer.addAll(PercentileHistogramBuckets.getDefaultBuckets(minTimerValue, maxTimerValue, true));
+
+                    Iterator<Double> itr = defaultBucketsMicrometer.iterator();
+
+                    while (itr.hasNext()) {
+                        buckets.put(itr.next(), new BucketValue(0, metadata.getUnit()));
+
+                    }
+                }
+
+            }
+
+        }
+
+        Optional<String> histogramOptionalData = ConfigProvider.getConfig().getOptionalValue("mp.metrics.distribution.histogram.buckets", String.class);
+        Optional<String> timerOptionalData = ConfigProvider.getConfig().getOptionalValue("mp.metrics.distribution.timer.buckets", String.class);
+
+        if (histogramOptionalData.isPresent() || timerOptionalData.isPresent()) {
+            HistogramBucketConfiguration bucketsConfig = MetricsConfigurationManager.getInstance().getHistogramBucketConfiguration(metricName);
+
+            TimerBucketConfiguration timerBucketsConfig = MetricsConfigurationManager.getInstance().getTimerBucketConfiguration(metricName);
+
+            if (metadata.getType().equals("histogram")) {
+                if (bucketsConfig != null && bucketsConfig.getValues() != null
+                    && bucketsConfig.getValues().length > 0) {
+                    double[] vals = Stream.of(bucketsConfig.getValues()).mapToDouble(Double::doubleValue).toArray();
+
+                    bucketArr = vals;
+
+                    for (Double value : bucketArr) {
+                        buckets.put(value, new BucketValue(0, metadata.getUnit()));
+                    }
+                    buckets.put(Double.POSITIVE_INFINITY, infiniteObject);
+
+                }
+            }
+
+            if (metadata.getType().equals("timer")) {
+                if (timerBucketsConfig != null && timerBucketsConfig.getValues() != null
+                    && timerBucketsConfig.getValues().length > 0) {
+                    Duration[] vals = timerBucketsConfig.getValues();
+
+                    for (Duration value : vals) {
+                        try {
+                            buckets.put(value.toMillis() / 1000.0, new BucketValue(0, metadata.getUnit()));
+                        } catch (Exception e) {
+                            e.printStackTrace();
+                        }
+
+                    }
+                    buckets.put(Double.POSITIVE_INFINITY, infiniteObject);
+                }
+            }
+
+        }
+
+        if (metadata != null && buckets != null && !buckets.isEmpty() && betaFenceCheck()) {
+            allBuckets.put(metricName, buckets);
+        }
+
+    }
+
+    private boolean betaFenceCheck() throws UnsupportedOperationException {
+        // Not running beta edition, throw exception
+        if (!ProductInfo.getBetaEdition()) {
+            if (!issuedBetaWarning) {
+                Tr.warning(tc, "This method is beta and is not available.");
```

Comment:
Not sure what the rules were, any messages not debug needs a code and translation?